toolchain/check/pattern_match.cpp

```diff
@@ -277,6 +281,7 @@ auto MatchContext::EmitPatternMatch(Context& context,
           pattern.loc_id, {.type_id = return_slot_pattern.type_id,
                            .type_inst_id = return_slot_pattern.type_inst_id,
                            .storage_id = entry.scrutinee_id});
+      results_.push_back(entry.scrutinee_id);
```

Comment:
For my own understanding, why is this pushed now but not before?

Reply:
Because we're now computing the calling-convention parameter list, and the return slot is part of the calling convention.

Previously we were computing it separately (see line 318 of the status quo) and exposing it separately (as `return_slot_id` in places like `Function` and `NameComponent`). With this PR, we're computing it together with the rest of the calling convention parameters, and exposing it both as a `return_slot_id` member and in the parameter list, as a transitional step toward removing the `return_slot_id` members. That cleanup will happen in the next PR, which as a byproduct will remove `return_slot_id_` from this class, making this line the only way this case produces output.

Reply:
Thanks!